### ISSUE TITLE
watsonx prompt-builder like parameters in generate()

### ISSUE THREAD

**Issue by florenzi002** [IBM/ibm-generative-ai#34]
**Context**
While working with the .py library there doesn't seem to be a way to submit request following the "prompt builder" pattern as available in the UI, below the payload from interaction with the webapp

```json
{
  "model_id": "",
  "inputs": [
    ""
  ],
  "template": {  <-------- this whole block cannot be set in the generate request through the library
    "id": "prompt_builder",
    "data": {
      "instruction": "",
      "input_prefix": "Input:",
      "output_prefix": "Output:",
      "examples": [
        {
          "input": "",
          "output": ""
        }
      ]
    }
  },
  "parameters": {
    "decoding_method": "greedy",
    "repetition_penalty": 1,
    "stop_sequences": [],
    "min_new_tokens": ,
    "max_new_tokens":
  }
}
```

the `template` param seems to be missing here https://github.com/IBM/ibm-generative-ai/blob/main/src/genai/services/request_handler.py

**Alternatives**
Submitting through the UI or possibly using th api directly, but that would defy the purpose of a library

**Has the feature been requested before?**

looks like there's a PR https://github.com/IBM/ibm-generative-ai/pull/27 but it is allowing for generic options and am not sure if it allows to pass "prompt-builder" queries to the API

**If the feature request is approved, would you be willing to submit a PR?**

Don't think I have the relevant knowledge here but could try to help

**Comment by vedem1192** [comment]
Hi @florenzi002 , thanks for opening the issue!
There is currently an open PR for this exact feature ([PR link](https://github.com/IBM/ibm-generative-ai/pull/27)). We are aiming to have it merge and available very soon. 

In you specific example, you would query the prompt_builder as follow : 
```
PromptPattern.from_watsonx(credentials=creds, id="prompt_builder")
```

**Comment by mirianfsilva** [comment]
Prompt-templating already available: [release v0.1.18](https://github.com/IBM/ibm-generative-ai/releases/tag/v0.1.18)
